// @node-red/nodes/core/fusion/a_sensor.js
module.exports = function(RED) {
    "use strict";

    function SensorNode(n) {
        RED.nodes.createNode(this, n);
        var node = this;
        this.location = n.location;
        this.device = n.device;
        this.devicetype = n.devicetype;
        this.sensingFunction = n.sensingFunction;

        var newMsg = {};
        var steps = Number(1);
        Object.assign(newMsg, {steps: steps});

        // 在节点初始化时自动执行一次
        setImmediate(function() {
            node.emit("input", {});
        });

        this.on("input", function(msg, send, done) {
            msg = msg || {};  // 防止 msg 未定义
            msg.type = node.type;
            msg.step = Number(1);
            msg.location = node.location;
            if (node.device) {
                msg.device = node.device;
            }
            if (node.devicetype) {
                msg.devicetype = node.devicetype;
            }
            msg.sensingFunction = node.sensingFunction;
            var nodeName = node.name;
            newMsg[nodeName] = msg;

            send(newMsg);
            done();
        });
    }

    RED.nodes.registerType("sensor", SensorNode);

}
