// @node-red/nodes/core/Service Composition/a_device_service.js
module.exports = function(RED) {
    "use strict";

    function DeviceServiceNode(n) {
        RED.nodes.createNode(this, n);
        var node = this;
        this.name = n.name;
        this.location = n.location;
        this.deviceName = n.deviceName;
        this.deviceType = n.deviceType;
        this.deviceId = n.deviceId;  // 添加 deviceId

        setImmediate(function() {
            node.emit("input", {});
        });

        this.on("input", function(msg, send, done) {
            msg = msg || {};
            msg.name = node.name;
            msg.type = node.type;
            msg.step = 1;
            msg.location = node.location;

            if (node.deviceName) {
                msg.deviceName = node.deviceName;
            }
            if (node.deviceType) {
                msg.deviceType = node.deviceType;
            }
            if (node.deviceId) {
                msg.deviceId = node.deviceId;  // 发送 deviceId
            }

            send(msg);
            done();
        });
    }

    RED.nodes.registerType("Device Service", DeviceServiceNode);
};
