// @node-red/nodes/core/Service Composition/a_device_service.js
module.exports = function(RED) {
    "use strict";

    function DeviceServiceNode(n) {
        RED.nodes.createNode(this, n);
        var node = this;
        this.name = n.name;
        this.location = n.location;
        this.device = n.device;
        this.deviceType = n.deviceType;
        this.ability = n.ability;

        var newMsg = {};
        Object.assign(newMsg, { step: 1 });

        setImmediate(function() {
            node.emit("input", {});
        });

        this.on("input", function(msg, send, done) {
            msg = msg || {};
            msg.name = node.name;
            msg.type = node.type;
            msg.step = 1;
            msg.location = node.location;
            if (node.device) {
                msg.device = node.device;
            }
            if (node.deviceType) {
                msg.deviceType = node.deviceType;
            }
            msg.ability = node.ability;

            send(msg);
            done();
        });
    }

    RED.nodes.registerType("Device Service", DeviceServiceNode);
};
