// @node-red/nodes/core/Event Fusion/a_sensor.js
module.exports = function (RED) {
    "use strict";

    function SensorNode(n) {
        RED.nodes.createNode(this, n);
        let node = this;

        this.location = n.location;
        this.sensorId = n.sensorId;
        this.deviceName = n.deviceName;
        this.deviceType = n.deviceType;
        this.sensingFunction = n.sensingFunction;

        let newMsg = {};
        let steps = 1;
        Object.assign(newMsg, {steps: steps});

        // 在节点初始化时自动执行一次
        setImmediate(function () {
            node.emit("input", {});
        });

        this.on("input", function (msg, send, done) {
            msg = msg || {};
            msg.step = 1;
            msg.type = node.type;
            msg.location = node.location;
            msg.sensorId = node.sensorId;

            // 使用与前端匹配的字段名输出
            if (node.deviceName) {
                msg.deviceName = node.deviceName;
            }
            if (node.deviceType) {
                msg.deviceType = node.deviceType;
            }
            msg.sensingFunction = node.sensingFunction;

            let nodeId = node.id;
            newMsg[nodeId] = msg;
            send(newMsg);
            done();
        });
    }

    RED.nodes.registerType("Sensor", SensorNode);
}
