// @node-red/nodes/core/Event Fusion/a_sensor.js
module.exports = function (RED) {
    "use strict";

    function SensorNode(n) {
        RED.nodes.createNode(this, n);
        var node = this;
        this.location = n.location;
        this.sensorId = n.sensorId;  // 确保 sensorId 被正确传递过来
        this.deviceName = n.deviceName;
        this.deviceType = n.deviceType;
        this.sensingFunction = n.sensingFunction;

        var newMsg = {};
        var steps = Number(1);
        Object.assign(newMsg, {steps: steps});

        // 在节点初始化时自动执行一次
        setImmediate(function () {
            node.emit("input", {});
        });

        this.on("input", function (msg, send, done) {
            msg = msg || {};
            msg.type = node.type;
            msg.step = Number(1);
            msg.location = node.location;

            // 添加 sensorId 到 msg 对象
            msg.sensorId = node.sensorId;

            // 使用与前端匹配的字段名输出
            if (node.deviceName) {
                msg.deviceName = node.deviceName;
            }
            if (node.deviceType) {
                msg.deviceType = node.deviceType;
            }
            msg.sensingFunction = node.sensingFunction;

            var nodeName = node.id;
            newMsg[nodeName] = msg;
            send(newMsg);
            done();
        });
    }

    RED.nodes.registerType("Sensor", SensorNode);
}
