// @node-red/nodes/core/Event Fusion/c_publish.js
module.exports = function (RED) {
    "use strict";

    function PublishNode(config) {
        RED.nodes.createNode(this, config);
        let node = this;
        node.rulename = config.rulename;
        this.on("input", function (msg, send, done) {
            msg.rulename = node.rulename;
            delete msg._msgid; // 删除 _msgid 字段
            let flowNodes = [];
            RED.nodes.eachNode(function(n) {
                flowNodes.push(n);
            });
            // 将节点数据转换为 JSON 字符串
            let flowJson = JSON.stringify(flowNodes, null, 2);
            const dataToSend = {
                ruleJson: msg,
                flowJson: flowJson
            };
            const postData = JSON.stringify(dataToSend);
            console.log(JSON.stringify(msg, null, 2));
            fetch('http://127.0.0.1:8080/api/fusion/uploadrule', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: postData
            })
        });
    }

    RED.nodes.registerType("Publish", PublishNode);
}
